### PR TITLE
feat(securityGroup*): Add support to set ipIngress.description and ge…

### PR DESCRIPTION
**Pull request by aravindmd**
Adding support to set `description` for ipIngress rules and also support generic `tags` map during `securityGroup` upsert operation

### CODE REVIEW

File: clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/ops/securitygroup/SecurityGroupIngressConverter.groovy [hunk 1]
```diff
@@ -61,9 +61,9 @@ class SecurityGroupIngressConverter {
     List<IpPermission> ipPermissions = description.ipIngress.collect { ingress ->
       IpPermission permission = new IpPermission(ipProtocol: ingress.ipProtocol, fromPort: ingress.startPort, toPort: ingress.endPort)
       if (ingress.cidr?.contains(':')) {
-        permission.ipv6Ranges = [new Ipv6Range().withCidrIpv6(ingress.cidr)]
+        permission.ipv6Ranges = [new Ipv6Range().withCidrIpv6(ingress.cidr).withDescription(ingress.description)]
       } else {
-        permission.ipv4Ranges = [new IpRange().withCidrIp(ingress.cidr)]
+        permission.ipv4Ranges = [new IpRange().withCidrIp(ingress.cidr).withDescription(ingress.description)]
       }
       permission
```
Comment (jeyrschabu) [inline clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/ops/securitygroup/SecurityGroupIngressConverter.groovy]: do it once here: `permission.withDescription(ingress.description)`

File: clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/description/UpsertSecurityGroupDescription.groovy [hunk 2]
```diff
@@ -30,6 +30,8 @@ class UpsertSecurityGroupDescription extends AbstractAmazonCredentialsDescriptio
 
   boolean ingressAppendOnly = false
 
+  Map tags
```
Comment (jeyrschabu) [inline clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/description/UpsertSecurityGroupDescription.groovy]: Any reason to not use  `com.amazonaws.services.ec2.model.Tag`?

Comment (jeyrschabu) [reply clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/description/UpsertSecurityGroupDescription.groovy]:
```
tags: [
    {
        class: "com.amazonaws.services.ec2.model.Tag",
        value: "0AAB,0AAF",
        key: "nobuObjectIds",
        nobuObjectIds: "0AAB,0AAF"
    },
    {
        class: "com.amazonaws.services.ec2.model.Tag",
        value: "aravinddjavalab-124",
        key: "Name",
        Name: "aravinddjavalab-124"
    }
]
```

File: clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/ops/securitygroup/SecurityGroupLookupFactory.groovy [hunk 3]
```diff
@@ -136,9 +136,12 @@ class SecurityGroupLookupFactory {
          */
         retrySupport.retry({
           CreateTagsRequest createTagRequest = new CreateTagsRequest()
-          createTagRequest.withResources(result.groupId).withTags([
-            new Tag("Name", description.name)
-          ])
+          Collection<Tag> tags = new HashSet()
```
Comment (jeyrschabu) [inline clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/ops/securitygroup/SecurityGroupLookupFactory.groovy]: This should just be upserting the nobu tags right?

File: clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/ops/securitygroup/SecurityGroupIngressConverter.groovy (hunk 1 above)
Comment (aravindmd) [reply clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/ops/securitygroup/SecurityGroupIngressConverter.groovy]: The description is for the `IPRange` , i don't see one for `permission`. I think i can leave it as is

File: clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/ops/securitygroup/SecurityGroupLookupFactory.groovy (hunk 3 above)
Comment (aravindmd) [reply clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/ops/securitygroup/SecurityGroupLookupFactory.groovy]: yes , that is correct.

File: clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/description/UpsertSecurityGroupDescription.groovy (hunk 2 above)
Comment (aravindmd) [reply clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/description/UpsertSecurityGroupDescription.groovy]: switched to use `com.amazonaws.services.ec2.model.Tag`

Comment (jeyrschabu) [reply clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/description/UpsertSecurityGroupDescription.groovy]: Per our discussion offline, we will leave this as a `Map<String, Object>`

**Pull request merged**